### ISSUE TITLE
Chronos: Support time series models in 3rd party libs

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/BigDL#4335]
**Warning: This issue will only consider python backend.**

This design is used to define how Chronos can help a 3rd party time series lib to improve the performance(both training and inferencing) and accuracy.

Basically, our users will have a more flexible workflow and could bring any model(defined by 3rd party libs such as `pytorch-forecasting` or even customized model by `pytorch` or `python-lightning`)

## Why we need this design?

Time series analysis is an evolving domain and we are having more and more models every month(even every week, every day).

- Add new models to `bigdl/chronos/model` is quite time-consuming since we need to define `Forecaster` , `AutoModel` and support in `AutoTSEstimator` . And we may meet some license problem.
- Chronos API is designed for easy-coding and non-expert users, we can not expose all options/choice in this API. This may bring some concern among our customers about the flexibility.

## What API we will support?

**Warning: Naming and namespace is totally TBD.**

We will support a "Trainer" and a "Predictor" in Chronos.

### Trainer

This trainer is basically a nano pytorch trainer, **but we may add some time-series specific improvement in future**.

```python
from bigdl.chronos.utils.pytorch import TSTrainer
from pytorch_forecasting.models.deepar import DeepAR  # Any 3rd parth model

model = DeepAR(...)
trainer = TSTrainer(...)
trainer.fit(model, data)
```

### Predictor

This predictor is used to accelerate inference of a time series model, we basically use the function in nano pytorch inferencing, **but we may add some time-series specific improvement in future**.

```python
from bigdl.chronos.utils.pytorch import TSPredictor

tspredictor = TSPredictor(model, ...)  # model is the model you just trained
tspredictor.quantize(...)
tspredictor.predict(data, quantize=...)
tspredictor.predict_with_onnx(data, quantize=...)
tspredictor.predict_with_openvino(data, quantize=...)
tspredictor.export_onnx(...)
tspredictor.export_openvino(...)
tspredictor.save(...)
tspredictor.load(...)
tspredictor.model  # fetch original model
```

## How will we support this design step by step

1. Provide examples/use cases for TFT and DeepAR in `pytorch-forecasting`. At this time we will only support a `TSTrainer`.
  - [x] Pre-experiment on TFT and DeepAR (with Qing). We had 5X(cpx,112core), 2X(clx, 48core) training speed-up for TFT and 2X(cpx,112core), 1.5X(clx, 48core) training speed-up for DeepAR.
  - [ ] TFT use-case [WIP]
  - [ ] DeepAR use-case [WIP]
  - [ ] User Guide
2. Support the `TSPredictor`
  - [ ] `TSPredictor`
  - [ ] examples (TFT, DeepAR)
  - [ ] User Guide

**Comment by TheaperDeng** [comment]
@shane-huang
